### PR TITLE
Beta/dark mode workspaces update

### LINKED ISSUE #331: Update NPM and Storybook

### LINKED ISSUE #352: Improve relation of three internal projects (Workspaces?)

### PR DESCRIPTION
**Pull request by isacoder**
### Description

closes #352
closes #331
- #352
- #331

### Implementation
Workspaces working with node 16 npm 8

Actions have been updated and tested
- Lint is only available in the ui-lib but could be added for example and storybook (some fixes necessary I notice it complains hehehe)
- Deploy 
- Publish
- Updated Readme files
- Storybook updated to 6.4.22 because 6.5 requires React 18

Automatic publish v19.0.0-beta.2 was reviewed with scorer-surveillance project and worked without issues.

**Pending**
- Agree versioning in packages and main package
- clean package.json's? 
- Update dependencies

Resources

Dedup for  react-refresh
https://github.com/storybookjs/storybook/issues/17049

Storybook update with webpack 5
https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#cra5-upgrade

Re-open your Visual Studio code to refresh the lint rules.

Steps for next PR's:

- Solve high risk dependencies.
- Dockerized it?
- Update the script of Icons repo for newer version?

### Screenshoots

Zepplin plugin is working in web just not in the app but I don't know why but I think is good enough

Screen Shot 2022-08-03 at 17 27 37
Screen Shot 2022-08-03 at 17 29 46

**Testing**
When downloading this branch, although the folders for storybook and example have been moved to packages folder, I noticed that they stayed in my local (basically duplicated files will be visible in your system unless you delete them manually ).

I have updated the Read Me files (there are two now)
The How to run is in the the development read me in the ui-lib folder.

There are more than 66,000 changes..
My recommendation for review is to check the next files

- README.md (main and ui-lib)
- Package.json files (Main, ui-lib, storybook, example)
- Github Workflow's under folder .github

Outside of that, anything you consider to read is a plus 👯

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,99 +1,27 @@
...
-    "type": "git",
-    "url": "git+ssh://git@github.com/future-standard/scorer-ui-kit.git"
-  },
-  "main": "dist/index.js",
-  "module": "dist/index.modern.js",
-  "source": "src/index.tsx",
-  "engines": {
-    "node": ">=10"
-  },
+  "version": "2.0.0",
```
Comment (JoshLipps) [inline package.json:3]: should this have a beta in it?

File: packages/ui-lib/package.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-router-dom": "^5.2.0",
+    "react-scripts": "^5.0.1",
+    "react-use-websocket": "^2.9.1",
+    "styled-components": "^5.3.5"
+  },
+  "files": [
+    "dist",
+    "link-ui-kit.sh"
```
Comment (JoshLipps) [inline packages/ui-lib/package.json:78]: remove this and the file

File: packages/ui-lib/package.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  "directories": {
+    "example": "example"
+  },
+  "bugs": {
+    "url": "https://github.com/future-standard/scorer-ui-kit/issues"
+  },
+  "homepage": "https://future-standard.github.io/scorer-ui-kit/#/",
+  "bin": {
+    "link-ui-kit": "link-ui-kit.sh"
+  }
```
Comment (JoshLipps) [inline packages/ui-lib/package.json:98]: remove this too

File: packages/ui-lib/package.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
+{
+  "name": "scorer-ui-kit",
+  "version": "1.9.0-beta.4",
```
Comment (JoshLipps) [inline packages/ui-lib/package.json:3]:
```suggestion
  "version": "2.0.0-beta.0",
```

File: packages/ui-lib/package.json [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    "test": "run-s test:unit test:lint test:build",
+    "test:build": "run-s build",
+    "test:lint": "eslint ./src --ext .ts,.tsx",
+    "test:unit": "cross-env CI=1 react-scripts test --env=jsdom",
+    "test:watch": "react-scripts test --env=jsdom",
+    "svgr": "svgr --typescript -d src/svg/ src/svg",
+    "predeploy": "cd ../example && npm run build && cd ../storybook && npm run build-storybook",
+    "deploy": "gh-pages -d ../example/build",
+    "actions-deploy": "gh-pages -d ../example/build",
+    "link": "npm link && npm link react && npm link react-dom && npm link react-router-dom && npm link styled-components",
```
Comment (JoshLipps) [inline packages/ui-lib/package.json:30]: we can remove this link

**Review by atomworks — APPROVED** [review]

**Comment by isacoder** [comment]
This PR was useful to review changes in Workspaces but this wont be merged to main

**Pull request closed without merging**